### ISSUE TITLE
PPFormulaNet training-loss bug

### ISSUE THREAD

**Issue by sharmax-vikas** [huggingface/transformers#47317]
### System Info

PPFormulaNet training-loss bug — ForCausalLMLoss incorrectly shifts labels in encoder-decoder model
The problem: PPFormulaNetForConditionalGeneration.forward() uses self.loss_function() (line 491 in modular_pp_formulanet.py) which maps to ForCausalLMLoss. ForCausalLMLoss internally shifts labels left by 1 (labels[..., 1:]), but PPFormulaNet is an encoder-decoder model where the logits are already aligned with labels — no shift is needed.
This means the model trains against misaligned labels, losing the last target token.
Where:
# modular_pp_formulanet.py, line 491
loss = self.loss_function(
    logits=logits, labels=labels, vocab_size=self.config.text_config.vocab_size, **kwargs
)

The fix (same pattern as Florence2 after #46898):
loss = self.loss_function(
    logits=logits, labels=labels, vocab_size=self.config.text_config.vocab_size,
    shift_labels=labels, **kwargs
)

Passing shift_labels=labels tells ForCausalLMLoss to skip its internal shift, keeping labels aligned with logits.
Context: Florence2 (which PPFormulaNet inherits from) was already fixed in PR #46898 with this same pattern, but PPFormulaNet overrides forward() so the fix didn't propagate. eliaghazal also flagged PPFormulaNet as potentially affected in PR #47090 discussion, but no one has filed or fixed it yet.

### Information

- [ ] The official example scripts
- [ ] My own modified scripts

### Tasks

- [ ] An officially supported task in the `examples` folder (such as GLUE/SQuAD, ...)
- [ ] My own task or dataset (give details below)

### Reproduction

.

### Expected behavior

.

**Comment by Tharunbilluri** [comment]
This is already fixed on \`main\` — PR #46903 (merged 2026-07-23) fixed exactly this bug as part of a broader encoder-decoder loss-alignment fix, closing #46901 (filed independently, describing the same double-shift). Current \`main\` already has:

\`\`\`python
loss = self.loss_function(
    logits=logits,
    labels=None,
    vocab_size=self.config.text_config.vocab_size,
    shift_labels=shift_labels if shift_labels is not None else labels,
    **kwargs,
)
\`\`\`

in both \`modular_pp_formulanet.py\` and the generated \`modeling_pp_formulanet.py\`, matching the fix this issue proposes. Looks like this and #46901 were filed around the same time describing the same root cause. Confirmed by building \`main\` from source and reading the current code — nothing left to do here. Suggest closing as a duplicate of #46901 / fixed by #46903.
